awsshell/wizard.py

```diff
@@ -19,6 +21,81 @@
 LOG = logging.getLogger(__name__)
 
 
+class ParamCoercion(object):
+    """This class coerces string parameters into the correct type.
+
+    By default this converts strings to numerical values if the input
+    parameters model indicates that the field should be a number. This is to
+    compensate for the fact that values taken in from prompts will always be
+    strings and avoids having to create specific interactions for simple
+    conversions or having to specify the type in the wizard specification.
+    """
+
+    _DEFAULT_DICT = {
+        'integer': int,
+        'float': float,
+        'double': float,
+        'long': long_type
+    }
+
+    def __init__(self, type_dict=_DEFAULT_DICT):
+        """Initialize a ParamCoercion object.
+
+        :type type_dict: dict
+        :param type_dict: (Optional) A dictionary of converstions. Keys are
+        strings representing the shape type name and the values are callables
+        that given a string will return an instance of an appropriate type for
+        that shape type. Defaults to only coerce numbers.
+        """
+        self._type_dict = type_dict
+
+    def coerce(self, params, shape):
+        """Coerce the params according to the given shape.
+
+        :type params: dict
+        :param params: The parameters to be given to an operation call.
+
+        :type shape: :class:`botocore.model.Shape`
+        :param shape: The input shape for the desired operation.
+
+        :rtype: dict
+        :return: The coerced version of the params.
+        """
+        return self._coerce(params, shape)
+
+    def _coerce(self, params, shape):
+        name = shape.type_name
+        if isinstance(params, dict) and name == 'structure':
+            return self._coerce_structure(params, shape)
```

Comment:
You also need to handle map and (maybe timestamp for completeness?) types. It might not hurt to look at the serializer and parser code in botocore if you need to do this as they have implementation of input/output to model traversers that you can copy.